entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -261,17 +266,33 @@ class EntityHandler(
         @RequestBody requestBody: Mono<String>
     ): ResponseEntity<*> {
         val userId = extractSubjectOrEmpty().awaitFirst()
+        val entityUri = entityId.toUri()
 
-        if (!entityService.exists(entityId.toUri()))
+        if (!entityService.exists(entityUri))
             throw ResourceNotFoundException("Entity $entityId does not exist")
-        if (!authorizationService.userCanUpdateEntity(entityId.toUri(), userId))
+        if (!authorizationService.userCanUpdateEntity(entityUri, userId))
             throw AccessDeniedException("User forbidden write access to entity $entityId")
 
         val body = requestBody.awaitFirst()
         val contexts = checkAndGetContext(httpHeaders, body)
 
-        entityService.updateEntityAttribute(entityId.toUri(), attrId, body, contexts)
+        val expandedBody = expandJsonLdFragment(body, contexts) as Map<String, List<Any>>
+        val expandedAttrId = expandJsonLdKey(attrId, contexts)!!
+        val expandedPayload = mapOf(expandedAttrId to expandedBody)
+        entityAttributeService.partialUpdateEntityAttribute(entityUri, expandedPayload, contexts)
 
+        val updatedEntity = entityService.getFullEntityById(entityId.toUri(), true)
+        entityEventService.publishEntityEvent(
+            AttributeUpdateEvent(
+                entityId = entityUri,
+                attributeName = attrId,
+                datasetId = getPropertyValueFromMapAsString(expandedBody, NGSILD_DATASET_ID_PROPERTY)?.toUri(),
+                operationPayload = compactAndStringifyFragment(expandedAttrId, expandedBody, contexts),
```

Comment:
You did this extra processing rather than just sending the input body to have the same payload structure as append/update entity attributes ?